### PR TITLE
add utility function to create flambe config from template

### LINKED ISSUE #12: Generating configs from templates
**Is your feature request related to a problem? Please describe.**

Yes. I'm trying to dynamically inject pathnames into a config, then run an experiment using that config. Out of the box, there is no clean way to do this.

**Describe the solution you'd like**

A function that accepts a path to a Jinja2-templated flambe config, an output path, and key:val pairs to inject.

**Describe alternatives you've considered**

Loading the config into memory with YAML or flambe-YAML tools, editing in memory, then writing a new config to disk.

**Additional context**

Here's what I've come up with:

```python
import os
import re

import jinja2


def generate_config_from_template(template_path, config_path, remove_comments=False, **template_kwargs):
    dirname = os.path.dirname(template_path)
    basename = os.path.basename(template_path)
    loader = jinja2.FileSystemLoader(searchpath=dirname)
    env = jinja2.Environment(loader=loader)
    template = env.get_template(basename)
    with open(config_path, 'w') as f:
        for line in template.render(**template_kwargs).split('\n'):
            if remove_comments:
                line = re.sub('# .*', '', line).rstrip()
            if line:
                f.write(line + '\n')
```

Where a config template might look like:

```yaml
post_process_preds: 'post_process_preds_ext'
---
!Experiment

name: ada-text-classification
pipeline:

  # stage 0 - Load the dataset object SSTDataset and run preprocessing
  0_dataset: !Foo
    train_path: {{ train_path }}
    test_path: {{ test_path }}
    transform:
      text: !TextField
      label: !LabelField

  # stage 1 - train the text classifier on the SSTDataset
  1_train: !Trainer
    dataset: !@ 0_dataset  # link back to the existing dataset
    train_sampler: !BaseSampler  # define a way of sampling dataset
    val_sampler: !BaseSampler
    model: !TextClassifier
      embedder: !Embedder
        embedding: !torch.Embedding  # automatically use pytorch classes
          num_embeddings: !@ 0_dataset.text.vocab_size  # reference vocab size
          embedding_dim: 300
        encoder: !PooledRNNEncoder
          input_size: 300
          rnn_type: lstm
          n_layers: !g [2]
          hidden_size: 256
      output_layer: !SoftmaxLayer
        input_size: !@ 1_train.model.embedder.encoder.rnn.hidden_size
        output_size: !@ 0_dataset.label.vocab_size
        take_log: false
    loss_fn: !torch.NLLLoss  # Use existing PyTorch negative log likelihood
    metric_fn: !torch.NLLLoss  # Used for validation set evaluation
    optimizer: !torch.Adam
      params: !@ 1_train.model.trainable_params  # Link to model parameters
    max_steps: 2  # Each step runs `iter_per_step` iterations
    iter_per_step: 2  # Eval and checkpoint every 50 iterations
  2_eval: !Evaluator
    dataset: !@ 0_dataset
    model: !@ 1_train.model
    metric_fn: !torch.NLLLoss
    output_path: {{ preds_path }}
    eval_sampler: !BaseSampler
    eval_data: test
  3_post_process_preds: !post_process_preds.PostProcessPreds
    preds_path: !@ 2_eval.output_path
    preds_id_path: {{ preds_id_path }}
    post_processed_preds_path: {{ post_processed_preds_path }}
    label_vocab: !@ 0_dataset.label.vocab
```

### PR DESCRIPTION
**Pull request by williamabrwolf**
Fixes #12

### CODE REVIEW

File: flambe/utils/__init__.py [hunk 1]
```diff
@@ -0,0 +1 @@
+__all__ = ['config']
```
Comment (jeremyasapp) [inline flambe/utils/__init__.py]: Could you add a direct import:

```
from flambe.utils.config import generate_config_from_template

__all__ = ['generate_config_from_template']
```

That way people can import from flambe.utils directly :)

File: flambe/utils/config.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import re
+
+import jinja2
+
+
+def generate_config_from_template(template_path: str,
+                                  config_path: str,
+                                  remove_comments: bool = False,
+                                  **template_kwargs):
+    """
```
Comment (jeremyasapp) [inline flambe/utils/config.py]: This is missing a proper docstring. Please follow the numpy style:

```
Parameters
-----------
template_path: str
    description
config_path: str
 ...
```

Please also add typing to template_kwargs (Dics[str, str]?)

**Comment by jeremyasapp** [comment]
@williamabrwolf awesome. Can you also just add template_kwargs to the docstring, and address the failing test related to bandit?

**Comment by williamabrwolf** [comment]
Done (🤞 that tests pass)

**Comment by iitzco-asapp** [comment]
@williamabrwolf , I would also add `jinja2` to the requirements file. Let's not assume that it's a dependency of some existing flambe dependency (`Flask` in this case)...because if at some point `Flask` removes it, then we are broken.

**Comment by jeremyasapp** [comment]
Good point @iitzco-asapp. Also seems like bandit is still complaining. I did a bit of research and it seems like you have to specify the extension in an Environment object: https://stackoverflow.com/questions/17257138/cant-disable-the-autoescape-in-jinja2

Also @williamabrwolf make sure your docstrings are 72 character max per line. Note that you can run all these tests locally by calling `tox` :)

**Comment by williamabrwolf** [comment]
Addressed.

**Pull request merged**